MoreCyclopsUpgrades/Managers/UpgradeManager.cs

```diff
@@ -228,6 +239,9 @@ internal void HandleUpgrades()
                 if (techTypeInSlot == TechType.None)
                     continue;
 
+                // triggering the add event on this cyclops
+                upgradeModuleEventHandler.onUpgradeModuleAdd.Trigger(new UpgradeInfo(Cyclops, techTypeInSlot));
```

Comment:
I understand how this might help, but this is only useful for pure "boolean" upgrades.  
This method won't work for stacking upgrades (like Speed modules and Solar chargers)  
or tiered upgrades (like Engine Efficiency modules).